test/relayclient/RelayClient.test.ts

```diff
@@ -87,6 +88,22 @@ contract('RelayClient', function (accounts) {
   let from: Address
   let data: PrefixedHexString
   let gsnEvents: GsnEvent[] = []
+  const cheapRelayerUrl = 'http://localhost:54321'
+
+  // register a very cheap relayer, so client will attempt to use it first.
+  async function registerCheapRelayer (relayHub: RelayHubInstance): Promise<void> {
```

Comment:
I learned a hard way that utility functions inside test files are harmful. They can not be exported and lead to so much duplication it eventually hurts. Especially if you only use this function once so far.

Reply:
what do you suggest? I prefer a helper function than put the code inline.
there's a chance that it will be reused, and in that case, it should be easy to export or move into util folder.
but since its used just once right now, there's no need to do it in advance.

Reply:
There is totally a need to do it in advance. It grows into a net of functions all relying on local, global variables, state and stuff. Then once you want to export it you have to get to some advanced copy-pasting. All functions should be imported from somewhere. Then it can be easily reused in different tests.

Reply:
but if you keep every one-use utility function in a central place, you end up with a pile of garbage you no longer remember if/when it is in use.
The time to refactor out is when it is used twice.